modelci/hub/profile_.py

```diff
@@ -0,0 +1,125 @@
+            if str(container.image).split('\'')[1] in ('mlmodelci/pytorch-serving:latest', 'mlmodelci/onnx-serving:latest'):
+                print(f"Container:{container.name} is already serving.")
+                return container.name
+        container = serve(self.model.saved_path, device=device, name=f'{self.model.engine}-profiler')
+        print(f"Container:{container.name} is now serving.")
+        return container.name
+
+    def diagnose(self, server_name: str, batch_size: int = None, device='cuda', timeout=30) -> DynamicProfileResult:
+        """Start diagnosing and profiling model.
+
+        Args:
+            server_name (str): to assign a name for the container you are creating for model profile
+            batch_size (int): Batch size.
+            device (str): Device name.
+            timeout (float): Waiting for docker container timeout in second. Default timeout period is 30s.
+        """
+        # Check server status
+
+        model_status = False
+        retry_time = 0  # use binary exponential backoff algorithm
+        tick = time.time()
+        while time.time() - tick < timeout:
+            if self.inspector.check_model_status():
+                model_status = True
+                break
+            retry_time += 1
+            # get backoff time in s
+            backoff_time = random.randint(0, 2 ** retry_time - 1) * 1e-3
+            time.sleep(backoff_time)
+
+        if not model_status:  # raise an error as model is not served.
+            raise ServiceException('Model not served!')
+
+        if batch_size is not None:
+            self.inspector.set_batch_size(batch_size)
+
+        result = self.inspector.run_model(server_name=server_name, device=device)
+
+        dpr = DynamicProfileResult(
+            device_id=result['device_id'],
+            device_name=result['device_name'],
+            batch=result['batch_size'],
+            memory=ProfileMemory(
+                total_memory=result['total_gpu_memory'],
+                memory_usage=result['gpu_memory_used'],
+                utilization=result['gpu_utilization'],
+            ),
+            latency=ProfileLatency(
+                inference_latency=result['latency'],
+            ),
+            throughput=ProfileThroughput(inference_throughput=result['total_throughput']),
+            ip=get_ip(),
+            create_time=result['completed_time'],
+        )
+        return dpr
+
+    def __auto_select_client(self):
+        import cv2
+        # according to the serving engine, select the right testing client.
+        # TODO: replace the input None data in each client with self-generated data.
+        serving_engine = self.model.engine
+        if serving_engine == Engine.NONE:
+            raise Exception(
+                'please choose a serving engine for the model')
+            # TODO How can we deploy to all available platforms if we don't know the engine?
+        img = cv2.imread('./modelci/hub/client/data/cat.jpg')
```

Comment:
Please use absolute address. You can derive the test image address from the current file path `__file__`. 
You may use `from pathlib import Path` to do this.